crates/extensions/tedge_mqtt_bridge/src/lib.rs

```diff
@@ -99,36 +405,39 @@ impl MqttBridgeActorBuilder {
             8883,
         );
         cloud_config.set_manual_acks(true);
+        cloud_config.set_clean_session(false);
```

Comment:
Is this required? Because I don't think that'll work for C8y as it expects the clean session to be true as documented [here](https://cumulocity.com/docs/device-integration/mqtt/#mqtt-clean-session)

Reply:
Yes we need to make sure the mqtt connection to the cloud uses a clean session, whilst using a non-clean session when connecting to the local broker.

Reply:
> Yes we need to make sure the mqtt connection to the cloud uses a clean session, whilst using a non-clean session when connecting to the local broker.

Correct, but unfortunate.
Using a [clean session](http://docs.oasis-open.org/mqtt/mqtt/v3.1.1/os/mqtt-v3.1.1-os.html#_Ref362965194) on each and every connection defeats the purpose of QoS 1 and 2.